src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -28,49 +32,68 @@ internal class SilentRequest : RequestBase
         {
             _silentParameters = silentParameters;
 
-            _brokerStrategyLazy = new Lazy<ISilentAuthRequestStrategy>(() => brokerStrategyOverride ?? new SilentBrokerAuthStrategy(this,
+            _brokerStrategyLazy = new Lazy<ISilentAuthRequestStrategy>(() => brokerStrategyOverride ?? new BrokerSilentStrategy(this,
                                                                                                serviceBundle,
                                                                                                authenticationRequestParameters,
                                                                                                silentParameters,
                                                                                                serviceBundle.PlatformProxy.CreateBroker(null)));
-            _clientStrategy = clientStrategyOverride ?? new SilentClientAuthStretegy(this, serviceBundle, authenticationRequestParameters, silentParameters);
+            _clientStrategy = clientStrategyOverride ?? new CacheSilentStrategy(this, serviceBundle, authenticationRequestParameters, silentParameters);
 
             _logger = authenticationRequestParameters.RequestContext.Logger;
         }
 
-        internal async override Task PreRunAsync()
-        {
-            await _clientStrategy.PreRunAsync().ConfigureAwait(false);
-        }
-
         protected override async Task<AuthenticationResult> ExecuteAsync(CancellationToken cancellationToken)
         {
+            await UpdateRequestWithAccountAsync().ConfigureAwait(false);
```

Comment:
The broker silent auth did not do this before. Are you aware of any potential side effects?

Reply:
It did, this was part of the dreaded "PreExecute" which is now history.

Reply:
The PreRunAsync for the broker strategy does nothing today. it was only for the client strategy. Ive also seen some issues with broker coming from this code.

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/5bca60568023bf239a1f5f4d02579ba7cad7365c/src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentBrokerAuthStrategy.cs#L93

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/5bca60568023bf239a1f5f4d02579ba7cad7365c/src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentClientAuthStrategy.cs#L38

Reply:
Ok, I'll have a look